main/src/ca/uwaterloo/flix/language/phase/Reducer.scala

```diff
@@ -176,15 +177,30 @@ object Reducer {
     * Companion object for [[LocalContext]].
     */
   private object LocalContext {
-    def mk(): LocalContext = LocalContext(mutable.ArrayBuffer.empty, 0)
+    def mk(purity: Purity): LocalContext = LocalContext(mutable.ArrayBuffer.empty, 0, Purity.isControlImpure(purity))
   }
 
   /**
     * A local non-shared context. Does not need to be thread-safe.
     *
     * @param lparams the bound variables in the def.
     */
-  private case class LocalContext(lparams: mutable.ArrayBuffer[LocalParam], var pcPoints: Int)
+  private case class LocalContext(lparams: mutable.ArrayBuffer[LocalParam], private var pcPoints: Int, private val isControlImpure: Boolean) {
+
+    /**
+      * Adds n to the private [[pcPoints]] field.
+      */
+    def addPcPoints(n: Int): Unit = {
```

Comment:
isn't this always 1? i'd prefer `addPcPoint()`

Reply:
ok